traitsui/editors/video_editor.py

```diff
@@ -40,45 +40,61 @@ class VideoEditor(BasicEditorFactory):
     aspect_ratio = AspectRatio()
 
     #: The name of a trait to synchronise with the audio muted state of
-    #: the video.  The referenced trait should be a Bool.
-    muted = Str(sync_value='from', sync_name='muted')
+    #: the video.
+    muted_name = Str()
+    #: True if the audio is muted, False otherwise
+    muted = Bool(sync_value='from', sync_name='muted_name')
 
     #: The name of a trait to synchronise with the audio volume of
-    #: the video player.  The referenced trait should be a Range(0.0, 100.0).
-    #: Values are on a logarithmic scale.
-    volume = Str(sync_value='from', sync_name='volume')
+    #: the video player.
+    volume_name = Str()
+    #: Audio volume on a logarithmic scale
+    volume = Range(0.0, 100.0, 50.0,
+                   sync_value='from', sync_name='volume_name')
 
     #: The name of a trait to synchronise with the playback rate of
-    #: the video.  The referenced trait should be a Float.
-    playback_rate = Str(sync_value='from', sync_name='playback_rate')
+    #: the video.
+    playback_rate_name = Str()
+    #: The playback speed of the video. Negative values are allowed but may not
+    #: be supported by the underlying implementation.
+    playback_rate = Float(sync_value='from', sync_name='playback_rate_name')
 
     #: The name of a trait to synchronise with the player's state.
-    #: The referenced trait should be a PlayerState.
-    state = Str(sync_value='both', sync_name='state')
+    state_name = Str()
+    #: The state (stopped, playing, paused) of the player
+    state = PlayerState(sync_value='both', sync_name='state_name')
 
     #: The name of a trait to synchronise with the player's position.
-    #: The referenced trait should be a Float representing time in seconds.
-    position = Str(sync_value='both', sync_name='position')
+    position_name = Str()
+    #: The current position, in seconds, in the video.
+    position = Float(sync_value='both', sync_name='position_name')
 
     #: The name of a trait to synchronise with the player's duration.
-    #: The referenced trait should be a Float representing time in seconds.
-    duration = Str(sync_value='to', sync_name='duration')
+    duration_name = Str()
+    #: Duration of the loaded video in seconds
+    duration = Float(sync_value='to', sync_name='duration_name')
```

Comment:
The duration is "read only" - ie. sync value is `"to"` so we don't need the ability to provide a default via the factory.  So this should be something like `duration = Float(sync_value='to', sync_name='duration')`, but if the name is an empty string we don't want to do anything, so we're not forcing the user to have a duration.

Reply:
OK, so one improvement to the code is that https://github.com/enthought/traitsui/blob/42a42d5564a2f2beb7ff0bdab932673eba9fd78d/traitsui/editor.py#L649 should probably have a clause that we don't try and setattr if `sync_value == "to"`.